internal/git/provider/github_provider.go

```diff
@@ -1,16 +1,112 @@
 package provider
 
-import "golang.org/x/net/context"
+import (
+	"fmt"
 
-type GitHubProvider struct{}
+	"github.com/go-logr/logr"
+	"github.com/jenkins-x/go-scm/scm"
+	"github.com/jenkins-x/go-scm/scm/driver/github"
+	"github.com/jenkins-x/go-scm/scm/factory"
+	"golang.org/x/net/context"
+)
 
-func (p GitHubProvider) ListPullRequests(ctx context.Context, repo Repository) (_ []PullRequest) {
-	panic("not implemented") // TODO: Implement
+type GitHubProvider struct {
```

Comment:
Does this need to be exported -- I would think the idea is to go through `New`, and not to see the implementations directly, is that right?

Reply:
In most cases yes, that code is kind of 1:1 copy from enterprise and I try to keep them in a similar shape, so it's easier to fix things if we notice something in one repo, at least for the first few months (that's not an old thing yet).